### ISSUE TITLE
Spurious warning: "[Cop] does not support SafeAutoCorrect parameter" when SafeAutoCorrect is a valid parameter for all cops

### ISSUE THREAD

**Issue by dduugg** [rubocop/rubocop#14981]
## Steps to reproduce the problem

Create a `.rubocop.yml` with `SafeAutoCorrect: false` for any cop that does not already have `SafeAutoCorrect` listed in its default configuration:

```yaml
# .rubocop.yml
Lint/UselessAssignment:
  SafeAutoCorrect: false
```

Run RuboCop against any Ruby file:

```
$ bundle exec rubocop test.rb
```

The same warning is produced with `Safe: false` for cops that do not have `Safe` in their default config.

## Expected behavior

No warning should be emitted. `SafeAutoCorrect` (and `Safe`) are documented, recognized configuration parameters that are valid for all cops. They appear in `CONFIG_CHECK_KEYS` and are actively read at runtime via `cop_config.fetch('SafeAutoCorrect', true)` in `autocorrect_logic.rb`.

## Actual behavior

```
Warning: Lint/UselessAssignment does not support SafeAutoCorrect parameter.

Supported parameters are:

  - Enabled
  - AutoCorrect
```

The warning fires for any cop that does not have `SafeAutoCorrect` explicitly listed in its entry in `config/default.yml`. Only the ~53 cops that already have `SafeAutoCorrect:` in `default.yml` are unaffected.

The root cause is in `config_validator.rb`. `COMMON_PARAMS` does not include `Safe` or `SafeAutoCorrect`, so `each_invalid_parameter` treats them as unknown when they are absent from a cop's default config entry:

```ruby
COMMON_PARAMS = %w[Exclude Include Severity inherit_mode AutoCorrect StyleGuide Details
                   Enabled Reference References].freeze

def each_invalid_parameter(cop_name)
  default_config = ConfigLoader.default_configuration[cop_name]
  @config[cop_name].each_key do |param|
    next if COMMON_PARAMS.include?(param) || default_config.key?(param)
    # SafeAutoCorrect is neither in COMMON_PARAMS nor in most cops' default_config,
    # so it falls through and triggers the warning.
    yield param, supported_params
  end
end
```

The fix is to add `Safe` and `SafeAutoCorrect` to `COMMON_PARAMS`.

## RuboCop version

```
$ bundle exec rubocop -V
1.85.0 (using Parser 3.3.10.2, rubocop-ast 1.49.0, analyzing as Ruby 2.7, running on ruby 3.4.6) [arm64-darwin24]
```
